python/hsfs/core/arrow_flight_client.py

```diff
@@ -0,0 +1,335 @@
+                    ) from e
+
+        return afs_error_handler_wrapper
+
+    def _get_dataset(self, descriptor):
+        info = self._connection.get_flight_info(descriptor)
+        reader = self._connection.do_get(self._info_to_ticket(info))
+        return reader.read_pandas()
+
+    @_handle_afs_exception
+    def read_query(self, query, query_str):
+        query_encoded = json.dumps(self._get_query_object(query, query_str)).encode(
+            "ascii"
+        )
+        descriptor = pyarrow.flight.FlightDescriptor.for_command(query_encoded)
+        return self._get_dataset(descriptor)
+
+    @_handle_afs_exception
+    def read_path(self, path):
+        descriptor = pyarrow.flight.FlightDescriptor.for_path(path)
+        return self._get_dataset(descriptor)
+
+    @_handle_afs_exception
+    def create_training_dataset(self, feature_view, tds_version=1):
+        if not self._is_enabled:
+            raise FeatureStoreException("Arrow Flight Service is not enabled.")
+        training_dataset_metadata = self._training_dataset_metadata_from_feature_view(
+            feature_view, tds_version
+        )
+        training_dataset_encoded = json.dumps(training_dataset_metadata).encode("ascii")
+        buf = pyarrow.py_buffer(training_dataset_encoded)
+        action = pyarrow.flight.Action("create-training-dataset", buf)
+        for result in self._connection.do_action(action):
+            return result.body.to_pybytes()
+
+    def _path_from_feature_view(self, feature_view, tds_version):
+        training_dataset_metadata = self._training_dataset_metadata_from_feature_view(
+            feature_view, tds_version
+        )
+        path = (
+            f"{training_dataset_metadata['featurestore_name']}_Training_Datasets/"
+            f"{training_dataset_metadata['name']}_{training_dataset_metadata['version']}"
+            f"_{training_dataset_metadata['tds_version']}/"
+            f"{training_dataset_metadata['name']}_{training_dataset_metadata['version']}/"
+            f"{training_dataset_metadata['name']}_{training_dataset_metadata['version']}.parquet"
+        )
+        full_path = f"/Projects/{training_dataset_metadata['featurestore_name']}/{path}"
+        return full_path
+
+    def _training_dataset_metadata_from_feature_view(self, feature_view, tds_version):
+        training_dataset_metadata = {}
+        training_dataset_metadata["name"] = feature_view.name
+        training_dataset_metadata["version"] = f"{feature_view.version}"
+        training_dataset_metadata["tds_version"] = f"{tds_version}"
+        training_dataset_metadata["query"] = self._get_query_object(
+            feature_view.query, feature_view.query.to_string()
+        )
+        training_dataset_metadata[
+            "featurestore_name"
+        ] = feature_view.query._left_feature_group.feature_store_name.replace(
+            "_featurestore", ""
+        )
+
+        return training_dataset_metadata
```

Comment:
It isn't a good practice to do it like this. The dictionary can easily have misspelled ids, class might be better. Furthermore, from what I see here, you only need something like a dictionary with the following keys: `path` and `query`. By just having these 2 keys it would be simplified and there would no longer be duplications between hsfs and flyingduck (for composing `path`).

Reply:
This is an object that I will send to the backend, having it as a dict makes it easier to send and decode on the receiver side.

Reply:
Instead of sending this object can't we just send path and query?

Reply:
Yes you are right, it might even make sense to not construct it manually but to use `trainingdataset.location` instead. However, I'd prefer to change it when we actually hook this functionality up to hsfs. Currently it is not used, so I'd rather simply remove it for now.